File: SchoolWebApp-API/Project.Core/Entities/CBE/Exams/ExamType.cs
using SchoolWebApp.Core.Entities.Shared;

namespace SchoolWebApp.Core.Entities.CBE.Exams
{
    public class ExamType : SettingsBase
    {
        public required string Abbreviation { get; set; }

        /// <summary>
        /// Whether this type's results appear on the report form, in a column of
        /// their own. Types tracked internally only - weekly marathons, mocks - are
        /// left off. Called Internal before it was renamed for clarity.
        /// </summary>
        public bool ShowOnReportForm { get; set; }

        /// <summary>
        /// Whether a term may hold more than one exam of this type, such as a weekly
        /// marathon. Off for the usual opening, mid-term and end-term exams, so each
        /// has one exam a term and one report-form column.
        /// </summary>
        public bool AllowMultiplePerTerm { get; set; }
        public List<SchoolExam> SchoolExams { get; set; } = new ();
    }
}

File: SchoolWebApp-Client/src/app/cbe/exams/models/exam-type.ts
import {ResourceModel} from '@/core/models/ResourceModel';

export class ExamType extends ResourceModel<ExamType> {
    public name?: string;
    public description?: string;
    public rank?: number;
    public abbreviation?: string;
    // Results appear on the report form, in a column of their own.
    public showOnReportForm?: boolean;
    // A term may hold more than one exam of this type (e.g. weekly marathons).
    public allowMultiplePerTerm?: boolean;

    constructor(model?: Partial<ExamType>) {
        super(model);
    }
}
